splash/render_options.py

```diff
@@ -28,21 +35,23 @@ def fromrequest(cls, request, max_timeout):
 
         # 1. GET / POST data
         data = {key: values[0] for key, values in request.args.items()}
-        if request.method == 'POST':
-            content_type = request.getHeader('content-type')
+        if request.method == b'POST':
+            content_type = request.getHeader(b'content-type')
             if content_type:
                 request.content.seek(0)
 
                 # 2. application/json POST data
-                if 'application/json' in content_type:
+                if 'application/json' in content_type.decode('utf-8'):
                     try:
-                        data.update(json.load(request.content, encoding='utf8'))
+                        content = request.content.read()
+                        data.update(json.load(
+                            StringIO(content.decode('utf-8')), encoding='utf8'))
```

Comment:
Why is StringIO needed here? 
If you have a content as string you can use json.loads. 
Why isn't json.load working?

This may not be a problem here, but io.StringIO is worse than cStringIO.StringIO because it makes a copy of the data.